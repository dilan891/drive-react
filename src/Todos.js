import "./card.css"
//import {useState,useEffect} from "react";

export default function Todos(){
    /*useEffect(()=>{
        fetch("/")
        .then(e => console.log(e))
        .catch(e => console.log(e))
    })*/
    const imagenes = ["imagen1","imagen2","imagen3","imagen4","iamgen5","imagen6"]
    const carpet = ["carpet1","carpet2"];
    return(
        <div className="card-content">
            {carpet.map(c=>{
                return(
                    <div key={c} className="card">
                        {c}
                    </div>
                )
            })}
            {imagenes.map(i=>{
                return(
               <div key={i} className="card">
                       <h1>{i}</h1>
                </div> 
            )})}
            
        </div>
    )
}