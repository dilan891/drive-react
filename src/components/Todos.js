import "../css/card.css"
import {Link} from "react-router-dom"
import {useState,useEffect} from "react";

export default function Todos(props){
    const [carpets,setcarpet] =  useState([]);//guarda un objeto con los detalles de cada carpeta
    const [archives,setarchives] = useState([])//guarda un objeto con los detalles de cada archivo
    
    function dataFech(){    //peticion al servidor para obtener las carpetas 
        fetch("http://192.168.20.203:4000/api/carpets")
        .then(e => e.json())
        .then(e => {
            setcarpet(e)
        })
        .catch(e => {
            console.log(e)
            setcarpet([{name: "carpet1",_id: "60454742f4a5194e0c511965"},{name: "carpet2", _id: 2}]);
        })
        // peticion de los archivos sin una carpeta asignada
        fetch("http://192.168.20.203:4000/api/archives",{
            method: "POST",
            body: JSON.stringify({id: "none"}),
            headers:{
                "Content-Type": "application/json"
            }
        })
        .then(data => data.json())
        .then(data =>{
            setarchives(data)
        })
        .catch(e => {
            console.log(e)
            setarchives([{name: "imagenE1",id: 45},{name: "imagenE2",id: 64}])
        })
        
    }

    useEffect(()=>{ 
        props.setID("none");
        dataFech() //matriz vacia para que no se ejecute un loop
    },[props.update])// eslint-disable-line react-hooks/exhaustive-deps

    return(
        <div>
            {props.modal}
            <div className="card-content">
            {carpets.map(c=>{ 
                return(
                    <Link  to={"/carpeta/"+c._id}>
                        <div key={c._id} className="card">
                            <h2>{c.name}</h2>
                        </div>
                    </Link>
                )
            })}
            {archives.map(i=>{
                return(
               <div className="card img">
                    <div key={i.id} className="view-img">
                        <img src="" height="160px" alt="no se ve :("></img>
                        <p>sadsacevavcsvarvjnoergnbasvnavnsjkdbnnfjsnojnckjbfdfdvfgnsjgnfagbtfesbngsfhsdgbjfahgjkdfsaglhgfdnsgbidiubsvihgsfuighdsfugh</p>
                    </div>
                    <div className="title-name">{i.name}</div>
                </div> 
            )})}
        </div>    
        </div>
    )
}
