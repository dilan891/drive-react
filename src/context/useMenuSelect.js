import React, {useReducer,createContext} from 'react'

//createcontext
const menuVisibility = {
    select:{
        visible: "hidden",
        position: "absolute",
        id: null,
        name: null,
        idToMove: null,
        nameToMove: null
    }}

export const Menu = createContext()

//reducer
const reducer=(state,action) =>{
        switch(action.type){
            case "active":
                return {
                    visible: "visible",
                    position: "relative",
                    id: action.payload.id,
                    name: action.payload.name
                }
            case "setID":
                return {
                    visible: state.visible,
                    position: state.position,
                    id: state.id,
                    name: state.name,
                    idToMove: action.payload.id,
                    nameToMove: action.payload.name
                }
            case "desactive":
                return state=menuVisibility.select
            default:
                throw new Error();
        }
    }

//context options
const UseMenuSelect = (props) => {

    const [state, dispatch] = useReducer(reducer, menuVisibility.select)

    const activeteSelect = (id,name = "error") =>{
        dispatch({
            payload: {id: id,name:name},
            type: "active"
        })
    }

    const setId = (id,name = "error")=>{
        dispatch({
            payload: {id: id,name: name},
            type: "setID"
        })
    }

    const desactivateSelect = () =>{
        dispatch({
            type: "desactive"
        })
    }

    return (
        <div>
            <Menu.Provider value={{
                visible: state.visible,
                position: state.position,
                name: state.name,
                nameToMove: state.nameToMove,
                activeteSelect: activeteSelect,
                desactivateSelect,
                setId
            }}>
                {props.children}
            </Menu.Provider>
        </div>
    )
}

export {UseMenuSelect}
