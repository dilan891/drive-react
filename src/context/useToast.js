/*
    cambia los booleanos toast a true para mostrar mensajes
*/
import React, { useReducer, createContext } from 'react'

/*
interface objetos{
    previus: boolean | null,  // guarda el id de la carpeta anterior
    failtoastOpen: boolean,
    newCarpetToast: boolean,
    moveToast: boolean
}*/

const toasts = {
    previus: null,  // guarda el id de la carpeta anterior
    failtoastOpen: false,
    newCarpetToast: false,
    moveToast: false
}

export const ToastsContext = createContext();

const UseToast = (props) =>{
    //Reducer setttings
    const reducer = (state, action) => {
        switch (action.type) {
            case "fail":
                return {
                    previus: null,
                    newCarpetToast: false,
                    failtoastOpen: true,
                    moveToast: false
                }
            case "carpetToast":
                return {
                    previus: null,
                    newCarpetToast: true,
                    failtoastOpen: false,
                    moveToast: false
                }
            case "moveToastActive":
                return {
                    previus: null,
                    failtoastOpen: false,
                    newCarpetToast: false,
                    moveToast: true
                }
            case "desactiveAll":
                return state = toasts
            case "setPreviusId":
                return {
                    previus: action.payLoad.id,
                    failtoastOpen: false,
                    newCarpetToast: false,
                    moveToast: false
                }
            default:
                throw new Error();
        }
    }

    //Reducer
    const [state, dispatch] = useReducer(reducer, toasts)

    //funciones    
    const failToast = () => {
        dispatch({
            type: "fail"
        });
        setTimeout(() => dispatch({ type: "desactiveAll" }), 4000);
    }

    const carpetToast = () => {
        dispatch({
            type: "carpetToast"
        })
        setTimeout(() => dispatch({ type: "desactiveAll" }), 3000);
    }

    const moveToastActive = () => {
        dispatch({
            type: "moveToastActive"
        })
        setTimeout(() => dispatch({ type: "desactiveAll" }), 3000);
    }

    //id anterior
    const previusId = (id) => {
        dispatch({
            payLoad: { id },
            type: "setPreviusId"
        })
    }

    const values = {
        failToastOpen: state.failtoastOpen,
        moveToast: state.moveToast,
        newCarpetToast: state.newCarpetToast,
        previus: state.previus,
        previusId,
        failToast,
        carpetToast,
        moveToastActive
    }

    return (
        <div>
            <ToastsContext.Provider value={values}>
                {props.children}
            </ToastsContext.Provider>
        </div>
    )
}

export { UseToast }
