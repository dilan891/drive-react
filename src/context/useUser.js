import {useReducer,createContext} from "react";

const User =  {
    userName: "none"
}

export const UserContext = createContext();

const UseUser = (props) => {
    const reducer  = (state, action)=>{
        switch(action.type){
            case "login":
                return{
                    ...state,
                    userName: action.payload.username
                }
            case "close":
                return User
            default:
                throw new Error();
        }
    } 

    const [state, dispatch] = useReducer(reducer,User);

    const newLogin = (userData) =>{
        dispatch({
            type: "login",
            payload: userData
        })
    }

    const close = () =>{
        dispatch({
            type: "close"
        })
    }

    const values = {
        username: state.userName, 
        newLogin,
        close
    }

    return (
        <div className="all">
            <UserContext.Provider value={values}>
                {props.children}
            </UserContext.Provider>
        </div>
    )

}

export { UseUser }
