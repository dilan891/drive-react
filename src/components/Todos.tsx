/*
    Este componente muestra todas las carpetas y archivos no asignados a ninguna carpeta
*/
import "../css/card.css";
import { Link } from "react-router-dom";
import React, { useState, useEffect, useContext } from "react";
import Options from "./options";
import { Archive } from "react-bootstrap-icons";
import { DataFetch, DataFetchArchives } from "./api/fetchApi"
import { ToastsContext } from "../context/useToast"
import Preview from "./previewArchive";

interface props {
    setID: (id: string) => void
}

let content: { name: string, _id: string, }[] = [{ name: "Juan", _id: "60454742f4a5194e0c511965" }, { name: "carpet2", _id: "2" }];
let archiver: { name: string, _id: string, type: string }[] = [{ name: "imagenE1", _id: "45", type: "img" }, { name: "imagenE2", _id: "64", type: "img" }]

const Todos: React.FC<props> = (props) => {
    const { previusId, updateToast } = useContext(ToastsContext);
    const [carpets, setcarpet] = useState<{ name: string, _id: string, }[]>([]);//guarda un objeto con los detalles de cada carpeta
    const [archives, setarchives] = useState<{ name: string, _id: string, type: string }[]>([])//guarda un objeto con los detalles de cada archivo
    const [refresh, setrefresh] = useState(0)

    const refreshData = () => {  //refresca la pagina por cada actualizacion de datos
        setrefresh(refresh + 1)
    }

    async function dataFech() {
        //peticion al servidor para obtener las carpetas no asignadas   
        const carpetData = await DataFetch();
        (carpetData == null) ? setcarpet(content) : setcarpet(carpetData);
        // peticion de los archivos sin una carpeta asignada
        const archiveData = await DataFetchArchives();
        (archiveData == null) ? setarchives(archiver) : setarchives(archiveData);
    }

    const idPreview = () => {
        previusId("none");
    }

    useEffect(() => {
        props.setID("none");//pasa el id none a el componente padre
        dataFech() //matriz vacia para que no se ejecute un loop
        return () => {
            setarchives([])
        }
    }, [refresh, updateToast])// eslint-disable-line react-hooks/exhaustive-deps

    return (
        <div>
            <div className="card-content">
                {carpets.map(c => {
                    return (
                        <div key={c._id} className="card img" >
                            <Link onClick={idPreview} className="view-img" to={"/carpeta/" + c._id}>
                                <div className="carpert-card">
                                    <Archive size={50} className="carpet-icon" />
                                    <h2>{c.name}</h2>
                                </div>
                            </Link>
                            <div className="descript">
                                <div className="title-name"></div>
                                <Options refresh={refreshData} id={c._id} name={c.name} type={"carpet"} />
                            </div>
                        </div>
                    )
                })}
                {archives.map(i => {
                    console.log(i)
                    return (
                        <div key={i._id} className="card img">
                            <div className="view-img">
                                <Preview type={i.type} ></Preview>
                            </div>
                            <div className="descript">
                                <div className="title-name">{i.name}</div>
                                <Options refresh={refreshData} id={i._id} name={i.name} type={"archive"} />
                            </div>
                        </div>
                    )
                })}
            </div>
        </div>
    )
}

export default Todos;
