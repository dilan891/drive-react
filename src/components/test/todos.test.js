import React from "react";
import Todos from "../Todos"
import { BrowserRouter as Router } from 'react-router-dom';
import { screen, render, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import { UseMenuSelect } from "../../context/useMenuSelect";
import { UseToast } from "../../context/useToast"
import { UseUser } from "../../context/useUser";


beforeEach(() => render(<UseUser><Router><UseMenuSelect><UseToast><Todos setID={((e) => { console.log(e) }).bind(this)} /></UseToast></UseMenuSelect></Router></UseUser>))
describe("todos menu", () => {
    it("fetch carpeta", async () => {
        await waitFor(() => { screen.getByRole("heading") })
        expect(screen.getByRole("heading")).toHaveTextContent("juan")
    });
    it("fecth archivos", async () => {
        await waitFor(() => screen.getByText(/archivo1/i))
        expect(screen.getByText(/archivo1/i)).toBeInTheDocument()
    });
    it("render img", async () => {
        await waitFor(() => { screen.getByRole("img") })
        expect(screen.getByRole("img")).toBeInTheDocument()
    });
    it("snapshot 1", async () => {
        const Component = renderer.create(<Router><UseMenuSelect><UseToast><Todos setID={((e) => { console.log(e) }).bind(this)} /></UseToast></UseMenuSelect></Router>)
        let tree = Component.toJSON();

        expect(tree).toMatchSnapshot();
    });
})
