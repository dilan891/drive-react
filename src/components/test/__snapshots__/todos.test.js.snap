// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos menu snapshot 1 1`] = `
<div>
  <div
    className="card-content"
  />
</div>
`;
